src/vivarium/framework/values.py

```diff
@@ -145,19 +147,32 @@ def name(self):
     def setup(self, builder):
         self.step_size = builder.time.step_size()
         builder.event.register_listener('post_setup', self.on_post_setup)
+        self.add_constraint = builder.lifecycle.add_constraint
+
+        builder.lifecycle.add_constraint(self.register_value_producer, allow_during=['setup'])
+        builder.lifecycle.add_constraint(self.register_rate_producer, allow_during=['setup'])
+        builder.lifecycle.add_constraint(self.register_value_modifier, allow_during=['setup'])
+        builder.lifecycle.add_constraint(self.get_value, allow_during=['setup', 'post_setup', 'population_creation',
+                                                                       'simulation_end', 'report'])
 
     def on_post_setup(self, event):
         # FIXME: This should raise an error, but can't due to risk effect generation
```

Comment:
is this fixme still true since we switched to explicit inclusion of risk effects?